src/AppProviders.tsx

```diff
@@ -50,15 +52,17 @@ export function AppProviders({ children }: ProvidersProps) {
                     <I18nProvider>
                       <WalletProvider>
                         <KeepKeyProvider>
-                          <ModalProvider>
-                            <TransactionsProvider>
-                              <AppProvider>
-                                <FoxEthProvider>
-                                  <DefiManagerProvider>{children}</DefiManagerProvider>
-                                </FoxEthProvider>
-                              </AppProvider>
-                            </TransactionsProvider>
-                          </ModalProvider>
+                          <ErrorBoundary FallbackComponent={ErrorPage}>
```

Comment:
this is the highest this can be placed in the tree without the error page itself crashing